libraries/AP_GSOF/AP_GSOF.cpp

```diff
@@ -169,18 +181,254 @@ void AP_GSOF::parse_vel(uint32_t a)
 
 void AP_GSOF::parse_dop(uint32_t a)
 {
-    // https://receiverhelp.trimble.com/oem-gnss/index.html#GSOFmessages_PDOP.html?TocPath=Output%2520Messages%257CGSOF%2520Messages%257C_____12
+    // https://receiverhelp.trimble.com/oem-gnss/gsof-messages-pdop.html
     // Skip pdop.
     dop.hdop = be32tofloat_ptr(msg.data, a + 4);
 }
 
 void AP_GSOF::parse_pos_sigma(uint32_t a)
 {
-    // https://receiverhelp.trimble.com/oem-gnss/index.html#GSOFmessages_SIGMA.html?TocPath=Output%2520Messages%257CGSOF%2520Messages%257C_____24
+    // https://receiverhelp.trimble.com/oem-gnss/gsof-messages-sigma.html
     // Skip pos_rms
     pos_sigma.sigma_east = be32tofloat_ptr(msg.data, a + 4);
     pos_sigma.sigma_north = be32tofloat_ptr(msg.data, a + 8);
     pos_sigma.sigma_up = be32tofloat_ptr(msg.data, a + 16);
 }
+
+void AP_GSOF::parse_ins_full_nav(uint32_t a)
+{
+    // https://receiverhelp.trimble.com/oem-gnss/gsof-messages-ins-full-nav.html
+    ins_full_nav.gps_week = be16toh_ptr(msg.data + a);
```

Comment:
This is really not what you're supposed to be using a packed structure like `ins_full_nav` for.

Assuming msg.data contains a valid INS_FULL_NAV message, you would simply do a `memcpy` here to take a copy of the packet.  Users of the values would do their own conversion on the way out - and you would *only* do that on very-much-internal functions which are populating other structures (there will typically only be one place that does that)

... in fact, most drivers wouldn't even bother keeping around the packet at all - they would just directly extract into their internal data structures on receipt of the message (i.e. in here, after casting msg.data into ins_full_nav_t

Reply:
So we can't memcpy because of the endianness. I was hoping ins_full_nav object already has the endianness conversion done so the values can be used as is. If you are proposing I instead do a memcpy on the buffer, and then pick through and convert endianness on all the fields, I can switch to do doing that, however, again, this is copying existing patterns. Do we need to change all the other AP_GSOF code in this PR?

Reply:
It's a bit of a contradiction in terms you're creating here.  You're populating a PACKED structure which will never go on the wire, and named somehwat like you'd name something you'd put on the wire.

You're also probably doing unnecessary work here as I bet we're not using all of these fields!

I would suggest populating local member variables from this message as it comes off the wire *or* doing the memcpy (an assignment will probably do, actually) into the persistent packet structure and then do the conversion on the way out.  We do both of these things in ArduPilot, but the former is less prone to endianness bugs.

Reply:
... having looked at this, we're making the situation worse in here in terms of mixing up on-the-wire structures and in-memory structures, and my suggestions above would make the code rather better (i.e. just `memcpy` and convert on the way out of the structure).

However, I can't insist on that in this PR - you're good to leave this as-is